### ISSUE TITLE
Handle np.inf or -np.inf returned by objective function

### ISSUE THREAD

**Issue by SimonBlanke** [SimonBlanke/Gradient-Free-Optimizers#4]
This issue is opened because of SimonBlanke/Hyperactive#10.

The user @gr8den reported a problem concerning the case if the objective function returns np.inf or -np.inf.

After I studied those cases I came to the following conclusions:
 - If the score is always -np.inf then GFO will return the very first parameter set as the best, since there was never any improvement of the score during the optimization run.
 - GFO must be able to handle np.inf, -np.inf and np.nan
 -  When there is no solution for the selected set of parameters the objective function should return np.nan instead of -np.inf

**Comment by SimonBlanke** [comment]
The first of the conclusions was solved in 79132df3 by setting the current and best position and score to the newest without checking if the score improved. This is only done in the very first iteration.

**Comment by SimonBlanke** [comment]
The requirement "GFO must be able to handle np.inf, -np.inf and np.nan" has been solved in a40d0ae693f7a19bd514a8dc86c524982cb5e7b0, be5baa968075beb98dceb5a3742dbc48825600e4 and 330f075ab21d2a8b850f8c68d06ccdd913a4970a. Tests have been added in 11a2b18a600933aeaf80369a87d98c3303ee5b85.

Adding this functionality was more difficult than expected. I had to add positions/score in the search tracker for non-nan and non-inf scores. Those could then be used by the hill climbing evaluation step and the samples for smbo, which cannot handle nan or inf.

This will not affect the "results"-attribute. Iterations with nan and inf will still occur in it.
